components/transfer/commons/transfer_row_display.lua

```diff
@@ -0,0 +1,427 @@
+				String.nilIfEmpty(transfer.extradata.toteamsectemplate),
+			},
+			roles = {
+				String.nilIfEmpty(transfer.role2),
+				String.nilIfEmpty(transfer.extradata.role2sec),
+			},
+		},
+		platform = HAS_PLATFORM_ICONS and self:_displayPlatform(transfer.extradata.platform) or nil,
+		displayDate = String.nilIfEmpty(transfer.extradata.displaydate) or date,
+		date = date,
+		wholeteam = Logic.readBool(transfer.wholeteam),
+		players = self:_readPlayers(transfers),
+		references = self:_getReferences(transfers),
+		confirmed = transfer.extradata.confirmed,
+		confidence = transfer.extradata.confidence,
+		isRumour = transfer.extradata.isRumour,
+	}
+end
+
+---@param platform string
+---@return string?
+function TransferRowDisplay:_displayPlatform(platform)
+	if not HAS_PLATFORM_ICONS then return end
+	if Logic.isEmpty(platform) then return '' end
+	return Platform._getIcon(platform) or ''
+end
+
+---@param transfers transfer[]
+---@return transferPlayer[]
+function TransferRowDisplay:_readPlayers(transfers)
+	return Array.map(transfers, function(transfer)
+		local extradata = transfer.extradata
+		return {
+			pageName = transfer.player,
+			displayName = String.nilIfEmpty(extradata.displayname) or transfer.player,
+			flag = transfer.nationality,
+			icons = {String.nilIfEmpty(extradata.icon), String.nilIfEmpty(extradata.icon2)},
+			faction = extradata.faction,
+			chars = extradata.chars,
+		}
+	end)
+end
+
+---@param transfers transfer[]
+---@return string[]
+function TransferRowDisplay:_getReferences(transfers)
+	local references = {}
+	Array.forEach(transfers, function(transfer)
+		Array.extendWith(references, TransferRef.fromStorageData(transfer.reference))
+	end)
+	references = TransferRef.makeUnique(references)
+
+	return Array.map(references, function(reference)
+		local refType = reference.refType
+		local link = reference.link
+		if refType == 'web source' then
+			return Page.makeExternalLink(Icon.makeIcon{
+				iconName = 'reference',
+				color = 'wiki-color-dark',
+			}, link)
+		elseif refType == 'tournament source' then
+			return Page.makeInternalLink(Abbreviation.make(
+				Icon.makeIcon{iconName = 'link', color = 'wiki-color-dark'},
+				'Transfer wasn\'t formally announced, but individual represented team starting with this tournament'
+			), link)
+		elseif refType == 'inside source' then
+			return Abbreviation.make(
+				Icon.makeIcon{iconName = 'insidesource', color = 'wiki-color-dark'},
+				'Liquipedia has gained this information from a trusted inside source'
+			)
+		elseif refType == 'contract database' then
+			return Page.makeExternalLink(Abbreviation.make(
+				Icon.makeIcon{iconName = 'transferdatabase', color = 'wiki-color-dark'},
+				'This transfer was not formally announced, ' ..
+					'but was revealed by a change in the LoL Esports League-Recognized Contract Database'
+			), 'https://docs.google.com/spreadsheets/d/1Y7k5kQ2AegbuyiGwEPsa62e883FYVtHqr6UVut9RC4o/pubhtml#')
+		end
```

Comment:
This is duplicated with Transfer References more or less, at least the text and types. Unify?

Reply:
tried to unify it a bit
problem is that the displays are quite different (icons vs refs)